### PR TITLE
[ENHANCEMENT] Added note for config_version in great_expectations.yml

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- Added note in `great_expectations.yml` file saying that config_version usually does not need to be modified.

### CODE REVIEW

File: great_expectations/data_context/templates.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 #   - Read our docs: https://docs.greatexpectations.io/en/latest/reference/data_context_reference.html#configuration
 #   - Join our slack channel: http://greatexpectations.io/slack
 
+# config_version tells Great Expectations whether the config file has been generated by GE version 0.8.0 or later.
+# note: This usually does not need to be changed.
```
Comment (Shinnnyshinshin) [inline great_expectations/data_context/templates.py]: @spbail If you have any suggestions on what this comment can include, please let me know

Comment (spbail) [reply great_expectations/data_context/templates.py]: Hm I think the fact that it's generated by 0.8.0 or later isn't really relevant and might just be confusing? It's more like "config_version represents the syntax version of this config file. It's auto-generated and usually does not need to be changed". Not sure if the term "syntax version" makes any sense haha.

Comment (Shinnnyshinshin) [reply great_expectations/data_context/templates.py]: agree that "version 0.8.0 ..etc etc" may be too much.. I was definitely on the fence as I was writing it :) Thanks for your suggestions!

**Comment by Shinnnyshinshin** [comment]
@spbail  how about this version?

File: great_expectations/data_context/templates.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/data_context/templates.py]: how about this version?

File: great_expectations/data_context/templates.py [hunk 2]
```diff
@@ -11,6 +11,8 @@
 #   - Read our docs: https://docs.greatexpectations.io/en/latest/reference/data_context_reference.html#configuration
 #   - Join our slack channel: http://greatexpectations.io/slack
 
+# config_version refers to the syntactic version of this config file, and is used in maintaining backwards compatibility
```
Comment (Shinnnyshinshin) [inline great_expectations/data_context/templates.py:14]: how about this version? @spbail

File: great_expectations/data_context/templates.py [hunk 3]
```diff
@@ -11,6 +11,8 @@
...
+# It is auto-generated and usually does not need to be changed.
```
Comment (Shinnnyshinshin) [inline great_expectations/data_context/templates.py:15]: @spbail how about this version?

Comment (spbail) [reply great_expectations/data_context/templates.py:15]: Yeah I like it!

**Review by spbail — APPROVED** [review]
I like it!

**Comment by Shinnnyshinshin** [comment]
Thanks!

**Pull request merged**